webextension/background.js

```diff
@@ -129,8 +127,75 @@ const Config = (function() {
       }
     }
 
-    load() {
+    maybeSendTelemetry(message) {
+      browser.study.sendTelemetry(message);
+    }
+
+    get shieldStudySetup() {
+      const pref = `extensions.${browser.runtime.id.split("@")[0]}.variation`;
+      return {
+        activeExperimentName: browser.runtime.id,
+        allowEnroll: true,
+        studyType: "shield",
+        telemetry: {
+          send: true,
+          removeTestingFlag: !this._testingMode,
+        },
+        endings: {
+          "user-disable": {
+            baseUrl: null,
+          },
+          expired: {
+            baseUrl: null,
+          },
+          dataPermissionsRevoked: {
+            baseUrl: null,
+            study_state: "ended-neutral",
+          },
+        },
+        logLevel: this._testingMode ? 30 : 0,
+        variationOverridePreference: pref,
+        weightedVariations: UIVariants.map(name => {
+          return {name, weight: 1};
+        }),
+        expire: {
+          days: 14,
+        },
+      };
+    }
+
+    _activateShield() {
+      if (this._shieldActivatedPromise) {
+        return this._shieldActivatedPromise;
+      }
+
+      this._shieldActivatedPromise = new Promise((resolve, reject) => {
```

Comment:
fancier than we do it in the example, because you catch all the throws.

Reply:
Heh. Examples can get away with it ;)

But I'm sure someone will eventually notice any browser console messages about uncaught promise rejections, so I prefer my live code to consider them all (I can't help but do the "defensive coding" dance, even for experiments).